Token layout tab reorganised

bubblobill opened the pull request:
### Identify the Bug or Feature request
resolves #5097

### Description of the Change
Cut down version of the changes in #5104

This PR makes no changes to the token class and does not implement independant scaling by axes or rotation of the token image.

### Possible Drawbacks
It is lame

### Documentation Notes
Moved token layout panel to its own tab

### Release Notes
Moved token layout panel to its own tab

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/RPTools/maptool/5360)
<!-- Reviewable:end -->

bubblobill committed:
Token class:
clean up some property names, e.g. isFlippedIso to flippedIso so that getter becomes isFlippedIso().

Unnecessary Token removed from property tokenOpacity. Now just opacity.

Reordered to keep layout props together.
Added new layout prop imageRotation

Updated tokenLayoutProp macro functions

bubblobill committed:
Merge remote-tracking branch 'origin/TokenLayout' into TokenLayout

bubblobill committed:
Updated TokenDto to handle refactored property names and new property imageRotation

bubblobill committed:
Updated Token again due to mysterious update overwriting it

bubblobill committed:
Many things coming together

bubblobill committed:
Merge remote-tracking branch 'origin/TokenLayout' into TokenLayout

bubblobill committed:
Forgot I hadn't finished hexes

bubblobill committed:
Changed all comments to block comments to make it easier to find commented code.
Commented code removed

bubblobill committed:
Corrected squares in createGridShape.

bubblobill committed:
Added image rotation property to layoutProps functions

bubblobill committed:
Sanity saving string concatenation

bubblobill committed:
Spotless indent

bubblobill committed:
Code cleanup

bubblobill committed:
Code cleanup

bubblobill committed:
Refactoring

bubblobill committed:
Refactoring

bubblobill committed:
Refactoring

bubblobill committed:
Refactoring

bubblobill committed:
Refactoring
Token drawing moved to TokenRenderer
Facing-arrow painting moved to FacingArrowRenderer
Some functionality moved to ImageUtils

bubblobill committed:
Facing-arrow painting moved to FacingArrowRenderer

bubblobill committed:
Code cleanup
Remove old layout panel

bubblobill committed:
Code cleanup
Icons embedded

bubblobill committed:
Spotless exception removed

bubblobill committed:
Merge branch 'develop' into TokenLayout

bubblobill committed:
Merge branch 'develop' into TokenLayout

bubblobill committed:
Fixed getScaledTokenImage returning incorrect scaling for free-size images

bubblobill committed:
cleanup and spotless

bubblobill committed:
cleanup and spotless

bubblobill committed:
Merge branch 'develop' into TokenLayout

bubblobill committed:
Merge branch 'RPTools:develop' into TokenLayout

bubblobill committed:
Merge branch 'RPTools:develop' into TokenLayout

bubblobill committed:
ImageUtil:
- removed unused method.
- changed log messages to warn
MathUtil:
- Removed generics
- Removed unused and duplicate functions
Token:
- reverted renamed properties

bubblobill committed:
ImageUtil:
- removed unused method.
- changed log messages to warn
MathUtil:
- Removed generics
- Removed unused and duplicate functions
Token:
- reverted renamed properties
Changed MT Function to setLayoutProps

bubblobill committed:
Fixed mapToRange function
Stopped image drag mouse movement rounding to zero
Reworked SpinnerSliderPaired
Removed most uses of MathUtil.doublePrecision because using numbers
instead of string format produces inequitable values.
Fixed rotation guide not showing for negative values.

kwvanderlinde committed:
Merge branch 'develop' into TokenLayout

bubblobill committed:
Spotless

bubblobill committed:
- Finished rename to setLayoutProps function (tested)

bubblobill committed:
changed access thingies

bubblobill committed:
added clamp to zoom calculation

bubblobill committed:
added isDouble

bubblobill committed:
Gave up on veto handling for stopping things reaching undesirable values
. Added floor and ceiling properties to the numeric model instead.
Major rework of SpinnerSliderPaired
Spotless and tweaks

bubblobill committed:
removed debug logging calls

bubblobill committed:
Refactor: createGridShape moved to Grid

bubblobill committed:
Spotless

bubblobill committed:
Refactor: rename TokenLocation to TokenPosition

bubblobill committed:
Refactor: made flip direction into an enum

bubblobill committed:
spot

bubblobill committed:
DTO fix missing proprty

bubblobill committed:
Nerfed version for intermediate release

kwvanderlinde commented on src/main/java/net/rptools/maptool/model/Token.java:
```diff
@@ -232,32 +236,29 @@ public enum Update {
   private double sizeScale = 1;
+  private double scaleX = 1;
+  private double scaleY = 1;
+  private double imageRotation = 0;
```
All the `imageRotation` plumbing (field, accessors, macro functions) will need to be removed as well.

bubblobill replied:
Dammit.. thought i had caught them all.

bubblobill replied:
Apparently not even close. 
I think all references to imageRotation have now been removed.

bubblobill committed:
Removed property token image rotation and associated references.
Fixed problem with slider not updating in SpinnerSliderPaired with
mouse wheel event

kwvanderlinde reviewed (changes requested):
I like how it looks. There's just one bug to chase down yet before merging.

When I set offsets, the selection box for the token is offset by the expected amount, but the token image is offset by twice that amount. E.g., if I place a token here on a 100x100 grid:
![image](https://github.com/user-attachments/assets/78aad74e-c033-4bc7-aa97-0b6f7381408d)

Then set the offsets to -50, -50, this is what happens
![image](https://github.com/user-attachments/assets/c67ebbed-ee8b-4719-bad5-096167db7899)

And this is what should happen, and which is what happens on 1.16.2:
![image](https://github.com/user-attachments/assets/7be3371d-49e2-4ef4-8e95-f45e8f9644f3)

bubblobill committed:
Keeping up appearances.

bubblobill commented:
> Then set the offsets to -50, -50, this is what happens 
![image](https://private-user-images.githubusercontent.com/7492219/435234792-c67ebbed-ee8b-4719-bad5-096167db7899.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUwNTI4MDAsIm5iZiI6MTc0NTA1MjUwMCwicGF0aCI6Ii83NDkyMjE5LzQzNTIzNDc5Mi1jNjdlYmJlZC1lZThiLTQ3MTktYmFkNS0wOTYxNjdkYjc4OTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDQxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA0MTlUMDg0ODIwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2RjM2M2OWE0NWY5MzMxMjYyN2IyYTYyMTYwMzAwZWEzMzNhYzlmODJkZjMyMzE1Mzk0NjZjZmFjYjkyNDhkNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.iHxan0TlIZc0jrzAOo1wgs-Sz2GHjh1WVAy-xRk6kJ8)

Which is mostly right because this ->

> And this is what should happen, and which is what happens on 1.16.2:
 ![image](https://private-user-images.githubusercontent.com/7492219/435235602-7be3371d-49e2-4ef4-8e95-f45e8f9644f3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUwNTI4MDAsIm5iZiI6MTc0NTA1MjUwMCwicGF0aCI6Ii83NDkyMjE5LzQzNTIzNTYwMi03YmUzMzcxZC00OWUyLTRlZjQtOGU5NS1mNDVlOGY5NjQ0ZjMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDQxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA0MTlUMDg0ODIwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODI3MDE5ZTMxMDgyZmMzYjNjNzM4MmQxZWEyNjkxNDA1NzU4MjU0ZTE3NWNkZjg4NTJlOTc0NDMwMTI0MDM3YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.cu2GXC5RnjHmSN16rnkdTvrVjlw4JJ50r7mPF__vhjo)

is wrong. What you should be seeing is this

![image](https://github.com/user-attachments/assets/eb41d329-0385-4b94-8dfb-fa1e82baf339)

The token layout should primarily be used for correcting the image placement within the cell, i.e. its apparent position. example, a little zoom and some offset makes this guy fill the space and leaves his arm out in the cold.:
![image](https://github.com/user-attachments/assets/67100716-93cd-4ca5-9e75-8d559d217318)

Changing the actual position of the token is not related to the layout.

Changing the actual position of the token relative to its footprint should be handled in footprints. Part of a whole bundle of follow-on improvements to go with implementing the new footprint stuff.

Check the new commit which leaves the selection marquee where it belongs.
